src/app/clusters/power-topology-server/power-topology-server.h

```diff
@@ -0,0 +1,84 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <lib/core/Optional.h>
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/AttributeAccessInterface.h>
+#include <lib/support/CommonIterator.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace PowerTopology {
+
+using namespace chip::app::Clusters::PowerTopology::Attributes;
+
+using chip::Protocols::InteractionModel::Status;
+
+class Delegate
+{
+public:
+    Delegate(EndpointId aEndpointId) : mEndpointId(aEndpointId) {}
+    virtual ~Delegate() = default;
+
+    virtual CHIP_ERROR const GetAvailableEndpointAtIndex(size_t index, EndpointId & endpointId) = 0;
+    virtual CHIP_ERROR const GetActiveEndpointAtIndex(size_t index, EndpointId & endpointId)    = 0;
```

Comment:
Should document what exact error code indicates "index out of range" (aka "end of list");

Also, what is the "const" there for?  Returning `const CHIP_ERROR` does not make sense.  Is the intent for this to be a const method on the delegate?

Reply:
OK, added comment. const was left over from this returning a vector.